numpy/__init__.pyi

```diff
@@ -3927,7 +3925,7 @@ class poly1d:
         k: None | _ArrayLikeComplex_co | _ArrayLikeObject_co = ...,
     ) -> poly1d: ...
 
-class matrix(ndarray[_ShapeType, _DType_co]):
+class matrix(ndarray[_ShapeType_co, _DType_co]):
```

Comment:
It might be kinda neat to narrow this one to `bound=tuple[int, int]`.
But feel free to ignore/resolve this if you don't want to; I wouldn't mind merging it like this :)